File: JavaToKotlin/app/src/main/java/io/dustin/apps/common/model/extension/CommentExtension.java
package io.dustin.apps.common.model.extension;

import com.querydsl.core.BooleanBuilder;
import com.querydsl.core.annotations.QueryDelegate;
import com.querydsl.core.annotations.QueryEntity;
import io.dustin.apps.board.domain.community.comment.model.Comment;
import io.dustin.apps.board.domain.community.comment.model.QComment;

@QueryEntity

public class CommentExtension {

    @QueryDelegate(Comment.class)
    public static BooleanBuilder indexByCountPagination(QComment comment, Long nextId) {
        BooleanBuilder builder = new BooleanBuilder();
        if(nextId != null) {
            builder.and(comment.id.lt(nextId));
        }
        return builder;
    }
}

File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/common/model/extension/CommentExtension.kt
package io.dustin.apps.common.model.extension

import com.querydsl.core.BooleanBuilder

@QueryEntity
object CommentExtension {
    @QueryDelegate(Comment::class)
    fun indexByCountPagination(comment: QComment?, nextId: Long?): BooleanBuilder? {
        val builder = BooleanBuilder()
        if (nextId != null) {
            builder.and(comment.id.lt(nextId))
        }
        return builder
    }
}
